strutil.py

```diff
@@ -0,0 +1,611 @@
+    :param quote:Every character in `quote` is regarded as a quote. Add a `\` prefix to make
+    an exception. Segment between the same quotes is preserved.
+    By default, `quote` is `'"\''`.
+
+    :param preserve: preserve the quote itself if `preserve` is `True`.
+    By default, `preserve` is `False`.
+    :return: a list of string.
+    """
+
+    if sep == quote:
+        raise ValueError('diffrent sep and quote is required')
+
+    if sep is None:
+        if len(line) == 0:
+            return []
+        line = line.strip()
+
+    rst = ['']
+    n = len(line)
+    i = 0
+    while i < n:
+        quote_s, quote_e, escape = _findquote(line[i:], quote)
+
+        if len(escape) > 0:
+            lines = []
+            x = 0
+            for e in escape:
+                lines.append(line[x:i + e])
+                x = i + e + 1
+            lines.append(line[x:])
+            line = ''.join(lines)
+            n = len(line)
+
+        if quote_s < 0:
+            sub = n
+        else:
+            sub = i + quote_s
+
+        if i < sub:
+            sub_rst = line[i:sub].split(sep)
+            if sep is None:
+                if line[sub - 1] in string.whitespace:
+                    sub_rst.append('')
+                if line[i] in string.whitespace:
+                    sub_rst.insert(0, '')
+
+            head = rst.pop()
+            sub_rst[0] = head + sub_rst[0]
+            rst += sub_rst
+
+        if quote_s < 0:
+            break
+
+        # discard incomplete
+        # 'a b"c'  ->  ['a']
+        if quote_e < 0:
+            rst.pop()
+            break
+
+        head = rst.pop()
+
+        if preserve:
+            head += line[i + quote_s:i + quote_e + 1]
+        else:
+            head += line[i + quote_s + 1:i + quote_e]
+
+        rst.append(head)
+        i += quote_e + 1
+
+    return rst
+
+
+def parse_colon_kvs(data):
+    data = tokenize(data, quote='"\'')
+
+    ret = {}
+    for buf in data:
+        if ':' not in buf:
+            raise ValueError('invalid arguments, arguments'
+                             'need key-val like: "k:v"')
+
+        k, v = buf.split(':', 1)
+
+        ret[k] = v
+
+    return ret
+
+
+def page(lines, max_lines=10, control_char=True, pager=('less',)):
+    """
+    Display `lines` of string in console, with a pager program (`less`) if too many
+    lines.
+
+    It could be used in a interactive tool to display large content.
+
+    It output strings directly to stdout.
+    :param lines: is `list` of lines to display.
+    :param max_lines: specifies the max lines not to use a pager.
+    By default it is 10 lines.
+    :param control_char: specifies if to interpret controlling chars, such as color char in terminal.
+    :param pager: specifies the program as a pager.
+    It is a list of command and argument.
+    By default it is `('less',)`.
+    :return: Nothing
+    """
+    if len(lines) > max_lines:
+        pp = {'stdin': subprocess32.PIPE,
```

Comment:
Using `subprocess` instead would be all right in python3.
subprocess32 is meant to fix a deadlock bug of python2-subprocess.